src/UI/Component/Symbol/Icon/Factory.php

```diff
@@ -75,4 +75,30 @@ public function standard($name, $aria_label, $size='small', $is_disabled = false
      * @return 	\ILIAS\UI\Component\Symbol\Icon\Custom
      **/
     public function custom($icon_path, $aria_label, $size='small', $is_disabled = false);
+
+    /**
+     * ---
+     * description:
+     *   purpose: >
+     *     ILIAS allows users to upload icons e.g. for Main bar items.
+     *     Those, in opposite to the standard icons, can be delivered inline as base64.
+     *   composition: >
+     *     Instead of setting a background image via CSS-class, an image-tag is
+     *     contained in the icons's div.
+     *   rivals:
+     *     Standard Icon: Standard Icons MUST be used for core-objects.
+     * rules:
+     *   style:
+     *     1: Inline Icons MUST use SVG as graphic, encoses as base64.
```

Comment:
@Amstutz: as long as only custom icons are implemented in the suggested way there is nothing against it. but like @akill mentioned in the JF the data-format should not be base64. svg is am XML-based format which can be used inline without any conversion or transcoding. standard icons should NOT be implemented as inline icons because of maintenance issues.